.github/workflows/pull-request.yml

```diff
@@ -107,6 +107,99 @@ jobs:
         with:
           path: './coverage/lcov.info'
           min_coverage: 95.0
+  
+  Health-Check:
+    name: Server health check
```

Comment:
- Instead of setting up the MongoDB and the repository again for this step, combine this step with the `Testing` workflow. 
- Make sure this step is run before the testing pipeline, as we do not want to test the application if the server doesn't start. 
- Add a similar test in `push.yml` as well.